domain-model/src/main/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculator.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.hmcts.cmc.domain.amount;
+
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.ClaimData;
+import uk.gov.hmcts.cmc.domain.models.Interest;
+import uk.gov.hmcts.cmc.domain.models.InterestDate;
+import uk.gov.hmcts.cmc.domain.models.amount.AmountBreakDown;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.util.Optional;
+
+import static java.math.BigDecimal.valueOf;
+import static java.util.Objects.requireNonNull;
+
+public class TotalAmountCalculator {
```

Comment:
From what I can see this class calculates other things in addition to total amounts. What do you think about changing the name to be less specific?

Reply:
`AmountCalculator`?